src/main/java/school/hei/haapi/endpoint/event/EventProducer.java

```diff
@@ -25,11 +26,12 @@
 
 @Component
 @Slf4j
-public class EventProducer implements Consumer<List<TypedEvent>> {
+public class EventProducer<T> implements Consumer<List<TypedEvent>> {
```

Comment:
Why do you need to generify EventProducer? In particular, you never use T in your implementation so it brings really nothing